### PR TITLE
Add definitions for restify's conditionalHandler plugin

### PR DESCRIPTION
**Pull request by leanazulyoro**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<http://restify.com/docs/plugins-api/#conditionalhandler>>
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/restify/index.d.ts [hunk 1]
```diff
@@ -946,6 +946,17 @@ export namespace plugins {
...
+    interface HandlerCandidate {
+        handler: RequestHandler | RequestHandler[];
+        version?: string | string[];
+        contentType?: string | string[];
+    }
+
+    /**
+     * Runs first handler that matches to the condition
+     */
+    function conditionalHandler(candidates: HandlerCandidate | HandlerCandidate[]): RequestHandler[];
```
Comment (weswigham) [inline types/restify/index.d.ts]: The [docs](http://restify.com/docs/plugins-api/#conditionalhandler) you've linked indicate that the function returns a single function - typo here in the `.d.ts`? Or bad docs? Also a test would be nice, if you're not opposed.

**Review by weswigham — CHANGES_REQUESTED** [review]

File: types/restify/index.d.ts (hunk 1 above)
Comment (leanazulyoro) [reply types/restify/index.d.ts]: It appears the docs are wrong, it actually returns a candidate object. I looked mostly at the implementation to get this right, here: https://github.com/restify/node-restify/blob/master/lib/plugins/conditionalHandler.js#LC107

I'll issue restify to fix the doc.
And I'll look into adding a test.

Comment (leanazulyoro) [reply types/restify/index.d.ts]: You're right, my bad. I'll fix that and see if I can a test

**Comment by cmincarelli** [comment]
@leanazulyoro - If you don't have time I can do this. (It would be really nice to have this included.) I think the code below should be enough for this to be valid. And as always, thanks for your additions! 

types/restify/index.d.ts

```typescript
/**
 * Runs first handler that matches to the condition
 */
function conditionalHandler(candidates: HandlerCandidate | HandlerCandidate[]): RequestHandler;
```
..and..

/types/restify/restify-tests.ts (around line 138~ish)
```typescript
server.use(restify.plugins.conditionalHandler([{
  contentType: ['text/plain'],
  handler: (req: restify.Request, res: restify.Response, next: restify.Next): void => {
	res.send('OK');
    next();
  },
  version: '0.0.0',
}]));
```

**Comment by leanazulyoro** [comment]
@cmincarelli yes i believe that's enought... I dont have time until the day after tomorrow to commit it myself... if you have the chance please go ahead and commit, so the pr doesnt get closed.

**Comment by leanazulyoro** [comment]
@cmincarelli, I've given you access to push to my forked repo.

**Comment by leanazulyoro** [comment]
Opened a new PR here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/25031
Sorry for letting this one abandoned.

**Pull request closed without merging**